### PR TITLE
See details for changes

### PR DESCRIPTION
**Pull request by jpiyali**
- Add support for web extension.
- code clean up
- Switch to using ICustomProperties interface for part C in apis.
- Add support for postfix for cookie name (no limits on cookie name length per browser spec.) from user configuration to avoid multiple sdk instances accessing same cookie for send buffer. This is not needed for user cookie as there is no bad side effect of having same anonymous user id generated for different instances of the sdk.

### CODE REVIEW

File: vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts [hunk 1]
```diff
@@ -104,27 +103,49 @@ export abstract class EnvelopeCreator {
...
+
+            if (web.domain) {
+                env.data.baseData.properties['domain'] =web.domain;
+            }
+
+            if (web.isManual) {
+                env.data.baseData.properties['isManual'] = web.isManual.toString();
+            }
+
+            if (web.screenRes) {
```
Comment (hectorhdzg) [inline vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts]: screenRes used to be in tags, is this change intentional?

File: vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts [hunk 2]
```diff
@@ -136,35 +157,26 @@ export abstract class EnvelopeCreator {
...
-        }
-        
         // No support for mapping Trace.traceState to 2.0 as it is currently empty
 
         if (item.ext.trace) {
             if (item.ext.trace.parentID) {
                 env.tags[CtxTagKeys.operationParentId] = item.ext.trace.parentID;
             }
+
+            if (item.ext.trace.name) {
```
Comment (hectorhdzg) [inline vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts]: What is this property and how is used?

File: vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts [hunk 3]
```diff
@@ -26,7 +26,7 @@ const durationProperty: string = "duration";
 export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IAppInsightsInternal {
     public static Version = "2.0.1-beta";
     public initialize: (config: IConfiguration, core: IAppInsightsCore, extensions: IPlugin[]) => void;
-    public identifier: string = "ApplicationInsightsAnalytics";
+    public identifier: string = "ApplicationInsightsAnalytics"; // do not change name or priority
```
Comment (hectorhdzg) [inline vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: Any particular reason the priority cannot be changed?

File: vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts (hunk 2 above)
Comment (jpiyali) [reply vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts]: I did not remove this, kept this mapping in for operation.name as it is auto captured in AI SDK today and I have not received confirmation that UI will work without it.

File: vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts (hunk 3 above)
Comment (jpiyali) [reply vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: when customers take dependency on our packages, priority changing can cause issues. Documentatioin for priority coming up today. No packages should update their priority unless there is a requirement

File: vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts (hunk 1 above)
Comment (jpiyali) [reply vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts]: https://1dsdocs.azurewebsites.net/articles/schema/common/schema-pageviews-and-usage.html screenRes is in web extension per documentation.

File: vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts (hunk 3 above)
Comment (hectorhdzg) [reply vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: Just wanted to understand if this plugin have any extra particular constraint

Comment (jpiyali) [reply vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts]: Got it, none. Its the same constraints for all plugins. Once finalized, dont change willy nilly.
